### PR TITLE
Allow ./configure to auto-bootstrap dependencies

### PR DESCRIPTION
**Pull request by edenhill**
.. and a bunch of build/packaging fixes to produce reliable builds, optionally statically linked.

Daughter projects should be adjusted to use `./configure --install-deps --source-deps-only ..` rather than any `apt-get install libssl-dev..` in their bootstrap scripts, since the latter is error prone.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,9 +109,42 @@ See the [wiki](https://github.com/edenhill/librdkafka/wiki) for a FAQ.
...
+On Windows, reference [librdkafka.redist](https://www.nuget.org/packages/librdkafka.redist/) NuGet package in your Visual Studio project.
+
+
+For other platforms, follow the source building instructions below.
+
+
 ### Building
 
       ./configure
+      # Or, to have configure automatically install dependencies:
```
Comment (ybyzek) [inline README.md]: "to have configure automatically install" reads strangely (at first I thought configure was a verb).  Perhaps:

```
# Or, to have `configure` automatically install dependencies:
```

or

```
# Or, to have it automatically install dependencies:
```

or

```
# Or, to automatically install dependencies:
```

File: README.md [hunk 2]
```diff
@@ -109,9 +109,42 @@ See the [wiki](https://github.com/edenhill/librdkafka/wiki) for a FAQ.
...
+      # ./configure --install-deps
+      # Or, let configure automatically build dependencies from source:
```
Comment (ybyzek) [inline README.md]: Same comment as above

File: README.md (hunk 1 above)
Comment (edenhill) [reply README.md]: 3rd option is good, thanks!

File: mklove/modules/configure.base [hunk 3]
```diff
@@ -304,6 +308,301 @@ function mkl_allvar_set {
...
+            cmd="yum install -y $pkgs"
+            ;;
+
+        alpine-0)
+            method=apk
+            pkgs=$(mkl_meta_get $name apk)
+            cmd="apk add $pkgs"
+            ;;
+
+        osx-*)
```
Comment (rnpridgeon) [inline mklove/modules/configure.base]: nit: It seems like sort of a stretch to cal brew MacOsx's `native` package installer.

File: mklove/modules/configure.base [hunk 4]
```diff
@@ -304,6 +308,301 @@ function mkl_allvar_set {
...
+
+# Returns 0 (yes) if this dependency has a source builder, else 1 (no)
+function mkl_dep_has_builder {
+    local name=$1
+    local func="${name}_install_source"
+    mkl_func_exists $func
+}
+
+
+# Returns 0 (yes) if this dependency has a source builder, else 1 (no)
```
Comment (rnpridgeon) [inline mklove/modules/configure.base]: has an installer*

File: mklove/modules/configure.base [hunk 5]
```diff
@@ -342,13 +641,27 @@ $(mkl_var_get MKL_FAIL__${n}__reason)
...
+            centos|rhel|redhat|fedora)
+                local rpms=$(mkl_meta_get $conf "rpm")
+                pkg_cmd="sudo yum install -y"
+                if [[ ${#rpms} > 0 ]]; then
+                    install_pkgs="$install_pkgs $rpms"
+                fi
+                ;;
+            alpine)
+                local apks=$(mkl_meta_get $conf "apk")
+                pkg_cmd="apk install "
```
Comment (rnpridgeon) [inline mklove/modules/configure.base]: Is there a `-y` equivalent for apk ?

File: mklove/modules/configure.cc [hunk 6]
```diff
@@ -49,7 +49,7 @@ function checks {
 	    mkl_mkvar_append CPPFLAGS CPPFLAGS "-m$MBITS"
 	    mkl_mkvar_append LDFLAGS LDFLAGS "-m$MBITS"
 	fi
-	if [[ -z "$ARFLAGS" && $MBITS == 64 && $MKL_DISTRO == "SunOS" ]]; then
+	if [[ -z "$ARFLAGS" && $MBITS == 64 && $MKL_DISTRO == "sunos" ]]; then
```
Comment (rnpridgeon) [inline mklove/modules/configure.cc]: Use `TR` to make case insensitive?

File: mklove/modules/configure.host [hunk 7]
```diff
@@ -51,12 +62,23 @@ function checks {
         esac
     fi
 
-    if [[ -z $sys ]]; then
+    # Convert to lower case
+    sys=$(echo $sys | tr '[:upper:]' '[:lower:]')
```
Comment (rnpridgeon) [inline mklove/modules/configure.host]: 🥇

File: mklove/modules/configure.base (hunk 3 above)
Comment (edenhill) [reply mklove/modules/configure.base]: I'm not sure I agree. Homebrew is the defacto package manager for developers on osx.
And it is native to the platform.

File: mklove/modules/configure.base (hunk 5 above)
Comment (edenhill) [reply mklove/modules/configure.base]: No, it is non-interactive it seems.
More importantly, there is no "install", it is called "add"

**Pull request merged**